File: packages/shared/src/types/extraction-status.ts
/** Mirrors `ExtractionStatus` in
 * `apps/api/src/modules/intelligence/domain/value_objects/extraction_status.py`. */
export enum ExtractionStatus {
  Pending = "PENDING",
  Processing = "PROCESSING",
  Completed = "COMPLETED",
  Failed = "FAILED",
  Confirmed = "CONFIRMED",
  Rejected = "REJECTED",
}

File: apps/api/src/modules/intelligence/domain/value_objects/extraction_status.py
from enum import StrEnum


class ExtractionStatus(StrEnum):
    """Lifecycle of one `DocumentExtraction`. `PROCESSING` exists for a
    future async pipeline (queues are explicitly out of scope this sprint —
    extraction runs synchronously inside the request, so callers only ever
    observe `PENDING` for an instant before `COMPLETED`/`FAILED`)."""

    PENDING = "PENDING"
    PROCESSING = "PROCESSING"
    COMPLETED = "COMPLETED"
    FAILED = "FAILED"
    CONFIRMED = "CONFIRMED"
    REJECTED = "REJECTED"
